packages/medusa/src/api/routes/admin/discounts/update-condition.ts

```diff
@@ -0,0 +1,96 @@
+import { IsOptional, IsString } from "class-validator"
+import { defaultAdminDiscountsFields, defaultAdminDiscountsRelations } from "."
+import { Discount } from "../../../../models/discount"
+import { DiscountService } from "../../../../services"
+import DiscountConditionService from "../../../../services/discount-condition"
+import { AdminUpsertConditionsReq } from "../../../../types/discount"
+import { getRetrieveConfig } from "../../../../utils/get-query-config"
+import { validator } from "../../../../utils/validator"
+/**
+ * @oas [post] /discounts/{id}/conditions/{condition_id}
+ * operationId: "PostDiscountsConditionsCondition"
+ * summary: "Updates a DiscountCondition"
+ * x-authenticated: true
+ * description: "Updates a DiscountCondition"
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       required:
+ *         - id
+ *       schema:
+ *         properties:
+ *           items:
+ *              properties:
+ *                products:
+ *                  type: array
+ *                  description: list of products
+ *                product_types:
+ *                  type: array
+ *                  description: list of product types
+ *                product_collections:
+ *                  type: array
+ *                  description: list of product collections
+ *                product_tags:
+ *                  type: array
+ *                  description: list of product tags
+ *                customer_groups:
+ *                  type: array
+ *                  description: list of customer_groups
+ * tags:
+ *   - Discount
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             discount:
+ *               $ref: "#/components/schemas/discount"
+ */
+
+export default async (req, res) => {
+  const { discount_id, condition_id } = req.params
+
+  const validated = await validator(
```

Comment:
wording here to, we when a method is long enough, you loose the context of what `validated` means.
We should come up with a common practice such as `validatedAdminPostDiscountsConditionsCondition`